tools/update-snapcraft

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+# Authors: Mert Kirpici <mert.kirpici@canonical.com>
+# Copyright (C) 2023 Canonical
+
+import logging
+from argparse import ArgumentParser
+from functools import cmp_to_key
+from pathlib import Path
+
+import feedparser
+import pygit2
+import semver
+import yaml
+
+logger = logging.getLogger(__name__)
+
+RELEASE_REPO_URL = 'https://opendev.org/openstack/releases.git'
+RELEASES_REPO_DIR = '/tmp/releases'
+OPENDEV_BASE_URL = 'git+https://opendev.org'
+OPENSTACK_REPO_URL_FMT = OPENDEV_BASE_URL + '/openstack/{project}.git@{ref}'
+OPENINFRA_REPO_URL_FMT = OPENDEV_BASE_URL + '/openinfra/{project}.git@{ref}'
+PYPI_RSS_FEED_FMT = 'https://pypi.org/rss/project/{project}/releases.xml'
+
+def parse_args():
+    """Parse command line arguments."""
+    parser = ArgumentParser()
+    parser.add_argument('release', help='OpenStack release')
+    return parser.parse_args()
+
+def get_latest_revision_from_release_file(path):
+    """Get the latest revision from yaml formatted release files."""
+    content = yaml.safe_load(path.read_text())
+    revisions = [release['version'] for release in content['releases']]
+    return f'{path.parent.stem}-last' \
+           if f'{path.parent.stem}-last' in revisions \
+           else sorted(revisions, key=cmp_to_key(semver.compare), reverse=True)[0]
```

Comment:
why not `max(...)`?

Reply:
because i hadn't thought of it :), implemented in the update. thanks.